subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c

```diff
@@ -2526,16 +2526,14 @@ int nrf_cloud_gnss_msg_json_encode(const struct nrf_cloud_gnss_data * const gnss
 	return 0;
 
 cleanup:
+	if (data_obj) {
+		cJSON_Delete(data_obj);
+	}
 	/* On failure, remove any items added to the provided object */
 	cJSON_DeleteItemFromObject(gnss_msg_obj, NRF_CLOUD_JSON_APPID_KEY);
 	cJSON_DeleteItemFromObject(gnss_msg_obj, NRF_CLOUD_JSON_MSG_TYPE_KEY);
 	cJSON_DeleteItemFromObject(gnss_msg_obj, NRF_CLOUD_JSON_DATA_KEY);
 	cJSON_DeleteItemFromObject(gnss_msg_obj, NRF_CLOUD_MSG_TIMESTAMP_KEY);
-	/* Cleanup data_obj if it wasn't added to gnss_msg_obj */
-	if (data_obj) {
-		cJSON_Delete(data_obj);
-	}
```

Comment:
Does this change functionality at all, or is it an aesthetic change? I don't mind it if it's the latter, just want to make sure I am following along

Reply:
i did not test it, but it seems like the former implementation would problem... the data item was deleted from `gnss_msg_obj` and then later `data_obj` deleted directly.

Reply:
Ah, yeah, I see it now -- though it seems like written this way you will have a similar problem, won't you? `cJSON_DeleteItemFromObject(gnss_msg_obj, NRF_CLOUD_JSON_DATA_KEY);` will cause a second call `cJSON_Delete(data_obj)`, which AFAICT has no protections against a double call like that

Reply:
i assumed that after directly deleting the data object, which could potentially be contained in the gnss object, that it would no longer be found in the gnss obj when it was searched for by key. i will confirm this tomorrow.

Reply:
my assumption was incorrect 😓 
the child obj needs to be detached from the parent before it is deleted (which `cJSON_DeleteItemFromObject` does).
updated, please review.